chat-commands.js

```diff
@@ -2479,6 +2479,60 @@ exports.commands = {
 	},
 	blacklisthelp: [`/blacklist [username], [reason] - Blacklists the user from the room you are in for a year. Requires: # & ~`],
 
+	battleban: function (target, room, user) {
+		if (!target) return this.parse('/help battleban');
+
+		target = this.splitTarget(target);
+		let targetUser = this.targetUser;
+		if (!targetUser) return this.errorReply(`User ${this.targetUsername} not found.`);
+		if (target.length > MAX_REASON_LENGTH) {
+			return this.errorReply(`The reason is too long. It cannot exceed ${MAX_REASON_LENGTH} characters.`);
+		}
+		if (!target) {
+			return this.errorReply("Battle banning require a reason.");
+		}
+		if (!this.can('lock', targetUser)) return;
+		if (targetUser.battlebanned) return this.errorReply(`User '${targetUser.name}' is already banned from battling.`);
+
+		const reasonText = target ? ` (${target})` : `.`;
+		const battlebanMessage = `${targetUser.name} was banned from starting new battles by ${user.name}${reasonText}`;
+		this.privateModAction(`(${battlebanMessage})`);
+
+		// Notify staff room when a user is banned from battling outside of it.
+		if (room.id !== 'staff' && Rooms('staff')) {
+			Rooms('staff').addByUser(user, `<<${room.id}>>${battlebanMessage}`);
+		}
+
+		this.globalModlog("BATTLEBAN", targetUser, ` by ${user.userid}${reasonText}`);
+		Ladders.cancelSearches(targetUser);
+		Punishments.battleban(targetUser, null, null, target);
+		targetUser.popup(`|modal|${user.name} has prevented you from starting new battles for 2 days${reasonText}`);
+		return true;
+	},
+	battlebanhelp: [`/battleban OR /bb [username], [reason] - Prevents the user from started new battles and shows them the [reason]. Requires: % @ * & ~`],
+
+	unbattleban: function (target, room, user) {
+		if (!target) return this.parse('/help unbattleban');
+		if (!this.can('lock')) return;
+
+		let targetUser = Users.get(target);
+		let reason = '';
+		if (targetUser && targetUser.battlebanned) {
+			reason = ' (`${targetUser.battlebanned}`)';
```

Comment:
there's a template string inside a regular string; should be  `` `(${targetUser.battlebanned})` ``